### PR TITLE
Add saving state for filters on dashboard [SCI-4984]

### PR DESCRIPTION
**Pull request by mlorb**
Jira ticket: [SCI-4984](https://biosistemika.atlassian.net/browse/SCI-4984)

### What was done
Add saving state for filters on dashboard.

### Note:
Fixes also setting to default state, when clicking "Clear" button.

### CODE REVIEW

File: app/assets/javascripts/dashboard/current_tasks.js [hunk 1]
```diff
@@ -46,7 +52,43 @@ var DasboardCurrentTasksWidget = (function() {
...
+  function filterStateSave() {
+    var teamId = $('.current-tasks-filters').data('team-id');
+    var filterState = {
+      sort: dropdownSelector.getValues(sortFilter),
+      statuses: dropdownSelector.getValues(statusFilter),
+      mode: $('.current-tasks-navbar .active').data('mode')
+    };
+
+    if (filterState) {
+      localStorage.setItem('current_tasks_filters_per_team/' + teamId, JSON.stringify(filterState));
```
Comment (aignatov-bio) [inline app/assets/javascripts/dashboard/current_tasks.js:67]: Maybe use prefix - `scinote_`

**Review by aignatov-bio — CHANGES_REQUESTED** [review]
Looks good, just few comments

File: app/assets/javascripts/dashboard/current_tasks.js (hunk 1 above)
Comment (okriuchykhin) [reply app/assets/javascripts/dashboard/current_tasks.js:67]: is it scoped to the origin?

File: app/assets/javascripts/dashboard/current_tasks.js [hunk 2]
```diff
@@ -46,7 +52,43 @@ var DasboardCurrentTasksWidget = (function() {
...
+  function filterStateLoad() {
+    var teamId = $('.current-tasks-filters').data('team-id');
+    var filterState = localStorage.getItem('current_tasks_filters_per_team/' + teamId);
+    var parsedFilterState;
+    var allStatusValues = $.map($(statusFilter).find('option'), function(option) {
+      return option.value;
+    });
+
+    if (filterState !== null) {
+      parsedFilterState = JSON.parse(filterState);
```
Comment (okriuchykhin) [inline app/assets/javascripts/dashboard/current_tasks.js:80]: What will happen if parsing throws an error?

File: app/views/dashboards/_current_tasks.html.erb [hunk 3]
```diff
@@ -29,7 +29,10 @@
             >
               <% MyModuleStatusFlow.find_each do |status_flow| %>
                 <% status_flow.my_module_statuses.each do |status| %>
-                  <option value="<%= status.id %>" <%= 'data-selected=true' unless status.final_status? %> ><%= status.name %></option>
+                  <option value="<%= status.id %>"
+                          data-completion-consequence="<%= status.my_module_status_consequences.where(type: "MyModuleStatusConsequences::Completion").any? %>">
```
Comment (aignatov-bio) [inline app/views/dashboards/_current_tasks.html.erb:33]: Do you test with addons? I think it will mark you only one status

**Review by aignatov-bio — CHANGES_REQUESTED** [review]
Check comments

File: app/views/dashboards/_current_tasks.html.erb (hunk 3 above)
Comment (mlorb) [reply app/views/dashboards/_current_tasks.html.erb:33]: I tested it. It marks first two statuses.

**Review by aignatov-bio — APPROVED** [review]

**Pull request merged**